lib/apiservers/engine/backends/container.go

```diff
@@ -1756,75 +1523,150 @@ func clientFriendlyContainerName(name string) string {
 }
 
-//This function is used to turn any call from docker create -v <stuff> into a volumeFields object.
-//the -v has 3 forms. 1: -v <anonymouse mount path>, -v <Volume Name>:<Destination Mount Path>, and -v <Volume Name>:<Destination Mount Path>:<mount flags>
-func processVolumeParam(volString string) (volumeFields, error) {
-	volumeStrings := strings.Split(volString, ":")
-	fields := volumeFields{}
+//------------------------------------
+// ContainerCreate() Utility Functions
+//------------------------------------
 
-	//This switch determines which type of -v was invoked.
-	switch len(volumeStrings) {
-	case 1:
-		VolumeID, err := uuid.NewUUID()
-		if err != nil {
-			return volumeFields{}, nil
-		}
-		fields.ID = VolumeID.String()
-		fields.Dest = volumeStrings[0]
-		fields.Flags = "rw"
-	case 2:
-		fields.ID = volumeStrings[0]
-		fields.Dest = volumeStrings[1]
-		fields.Flags = "rw"
-	case 3:
-		fields.ID = volumeStrings[0]
-		fields.Dest = volumeStrings[1]
-		fields.Flags = volumeStrings[2]
-	default:
-		//NOTE: the docker cli should cover this case. This is here for posterity.
-		return volumeFields{}, fmt.Errorf("Volume bind input is invalid : -v %s", volString)
-	}
-	return fields, nil
+// createInternalVicContainer() creates an container representation (for docker personality)
+// This is called by ContainerCreate()
+func createInternalVicContainer(image *metadata.ImageConfig, config *types.ContainerCreateConfig) (*viccontainer.VicContainer, error) {
+	// provide basic container config via the image
+	container := viccontainer.NewVicContainer()
+	container.ID = image.ID
+	container.Config = image.Config //Set defaults.  Overrides will get copied below.
+
+	return container, nil
 }
 
-func processAnonymousVolumes(h *string, volumes map[string]struct{}, client *client.PortLayer) ([]volumeFields, error) {
-	var volumeFields []volumeFields
+// SetConfigOptions is a place to add necessary container configuration
+// values that were not explicitly supplied by the user
+func setCreateConfigOptions(config, imageConfig *containertypes.Config) {
+	// Overwrite or append the image's config from the CLI with the metadata from the image's
+	// layer metadata where appropriate
+	if len(config.Cmd) == 0 {
+		config.Cmd = imageConfig.Cmd
+	}
+	if config.WorkingDir == "" {
+		config.WorkingDir = imageConfig.WorkingDir
+	}
+	if len(config.Entrypoint) == 0 {
+		config.Entrypoint = imageConfig.Entrypoint
+	}
+
+	// set up environment
+	setEnvFromImageConfig(config, imageConfig)
+}
 
-	for v := range volumes {
-		fields, err := processVolumeParam(v)
-		log.Infof("Processed Volume arguments : %#v", fields)
-		if err != nil {
-			return nil, err
+func setEnvFromImageConfig(config, imageConfig *containertypes.Config) {
+	// Set PATH in ENV if needed
+	setPathFromImageConfig(config, imageConfig)
+
+	containerEnv := make(map[string]string, len(config.Env))
+	for _, env := range config.Env {
+		kv := strings.Split(env, "=")
+		containerEnv[kv[0]] = kv[1]
```

Comment:
I think its also legit to not have an '=', unless there is some validation prior to reaching this point we should do:
```go
var val string
if len(kv) == 2 {
    val = kv[1]
}
containerEnv[kv[0]] = val
```